bgpd/bgp_nht.c

```diff
@@ -707,9 +707,37 @@ static void bgp_nht_ifp_table_handle(struct bgp *bgp,
 				     struct interface *ifp, bool up)
 {
 	struct bgp_nexthop_cache *bnc;
+	struct nexthop *nhop;
+	uint8_t other_nh_count;
+	bool nhop_found;
 
 	frr_each (bgp_nexthop_cache, table, bnc) {
-		if (bnc->ifindex_ipv6_ll != ifp->ifindex)
+		other_nh_count = 0;
+		nhop_found = false;
+		for (nhop = bnc->nexthop; nhop; nhop = nhop->next) {
+			if (nhop->ifindex != ifp->ifindex) {
+				other_nh_count++;
+				continue;
+			}
+			if (nhop->vrf_id == ifp->vrf->vrf_id) {
```

Comment:
If ifindex is the same and vrf_id is the same, then it means other_nh_count?

Reply:
you are right. I made an error when bringing those changes..